app/helpers/draws_helper.rb

```diff
@@ -22,9 +22,50 @@ def diff_class(diff)
     'zero'
   end
 
+  # Return the link to lock or unlock a specific group size for a draw
+  def toggle_size_lock_btn(draw:, size:, path:)
+    return lock_size_btn(draw, size, path) unless draw.size_locked?(size)
+    unlock_size_btn(draw, size, path)
+  end
+
+  # Return the link/button to proceed from the pre-lottery phase. Takes draw
+  # oversubscription status into account and modifies text and class as
+  # necessary.
+  #
+  # @param draw [Draw] the draw
+  # @return [String] the link
+  def proceed_from_pre_lottery_btn(draw)
+    if draw.oversubscribed?
+      btn_txt = 'Handle oversubscription'
+      options = { class: 'button alert' }
+      target = oversub_draw_path(draw, to_lottery: 1)
+    else
+      btn_txt = 'Proceed to lottery'
+      options = { class: 'button', method: :patch }
+      target = start_lottery_draw_path(draw)
+    end
+    link_to btn_txt, target, **options
+  end
+
   private
 
   def day_str(n)
     "#{n} #{'day'.pluralize(n)}"
   end
+
+  def lock_size_btn(draw, size, path)
+    link_to "Lock #{size_str_proper(size)}",
+            toggle_size_lock_draw_path(draw, size, redirect_path: path),
+            method: :patch, id: "lock-size-#{size}", class: 'button'
+  end
```

Comment:
can we add a descriptive tooltip ("Prevent students from forming more groups of this size")?